stdlib/map.mli

```diff
@@ -86,6 +86,16 @@ module type S =
        of [x] in [m] disappears.
        @before 4.03 Physical equality was not ensured. *)
 
+    val update: key -> ('a option -> 'a) -> 'a t -> 'a t
```

Comment:
The documentation already says that (same phrasing as add documentation).
The current implementation is really copied-pasted from add, but f is called to compute data, instead of taking it as an argument.